pinot-core/src/main/java/org/apache/pinot/core/segment/processing/collector/CollectorConfig.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.processing.collector;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+
+/**
+ * Config for Collector
+ */
+@JsonDeserialize(builder = CollectorConfig.Builder.class)
+public class CollectorConfig {
+  private static final CollectorFactory.CollectorType DEFAULT_COLLECTOR_TYPE = CollectorFactory.CollectorType.CONCAT;
+
+  private final CollectorFactory.CollectorType _collectorType;
+  private final Map<String, ValueAggregatorFactory.ValueAggregatorType> _aggregatorTypeMap;
+
+  private CollectorConfig(CollectorFactory.CollectorType collectorType,
```

Comment:
It should also include sorted columns (list of columns, where records are sorted on the first column firstly, then second column etc.)

Reply:
Added the config and support for sorting in the collection.
Kept the implementation very simple right now. Since we already have all records in memory due to aggregation step, simply called list.sort.
Lmk if you think any optimizations are needed there